### ISSUE TITLE
Exposing internal prometheus metrics

### ISSUE THREAD

**Issue by mwarzech21** [osrg/gobgp#2857]
What are the reasons for keeping the Prometheus metrics inside the internal package?
Would it be possible, to move them out from internal folder, in order to give access for gathering those metrics , while using only some specific gobgp parts and features?

**Comment by kacpekwasny** [comment]
+1

**Comment by fujita** [comment]
It means that it's more difficult to keep the compatibility because the internal might be changed in every release?

**Comment by SkalaNetworks** [comment]
Usually you try not to change prometheus metrics (at least their name and structure) in a way that isn't backward compatible. Even for internal packages, this would is the rule.

The reason is simple, modifying the name or deleting labels on a metric breaks panels and Prometheus rules.
There's probably nothing wrong with this package going public.

**Comment by mwarzech21** [comment]
Yes, thats what I was thinking. And unfortunately now, when those metrics are in the internal pkg, I cannot use them in my project.

Heres the PR for this issue I made a while ago:
https://github.com/osrg/gobgp/pull/2858

**Comment by piotrsuchy** [comment]
This would be useful for me as well. Looking forward to get it merged!
